ergo-lib/src/wallet/ext_secret_key.rs

```diff
@@ -0,0 +1,267 @@
+//! Extended private key operations according to BIP-32
+use std::convert::TryInto;
+
+use super::{
+    derivation_path::{ChildIndex, ChildIndexError, DerivationPath},
+    ext_pub_key::ExtPubKey,
+    mnemonic::MnemonicSeed,
+};
+use crate::ArrLength;
+use ergotree_interpreter::sigma_protocol::private_input::DlogProverInput;
+use ergotree_ir::{
+    serialization::{SigmaParsingError, SigmaSerializable, SigmaSerializationError},
+    sigma_protocol::sigma_boolean::ProveDlog,
+};
+use hmac::{Hmac, Mac, NewMac};
+use sha2::Sha512;
+use thiserror::Error;
+
+/// Private key (serialized Scalar) bytes
+pub type SecretKeyBytes = [u8; 32];
+/// Chain code bytes
+pub type ChainCode = [u8; 32];
+
+type HmacSha512 = Hmac<Sha512>;
+
+/// Result type for ExtendedSecretKey operations
+pub type Result<T> = std::result::Result<T, ExtSecretKeyError>;
```

Comment:
I'd refrain from this. It's a trade-off between speed of writing and readability and I find readability more important.